features/step_definitions/delivery_methods_steps.rb

```diff
@@ -36,7 +36,7 @@
 end
 
 Then(/^the customer should be notified by email$/) do
-  last_job = ActiveJob::Base.queue_adapter.enqueued_jobs.last
+  last_job = ActiveJob::Base.queue_adapter.enqueued_jobs.reverse.detect { |job| job[:job] == NotifyViaEmail }
```

Comment:
I can change the order the jobs are created in and then still use last. but that feels fragile. That is why I went with the approach of getting the most recent job of the required type.